core/src/test/scala/org/apache/spark/sql/delta/DeltaWriteConfigsSuite.scala

```diff
@@ -0,0 +1,486 @@
+  |table          |options_and_tblproperties|c_or_r_replace|true     |false                    |true                  |true                            |true                         |
+  |table          |options_and_tblproperties|c_or_r_replace|false    |false                    |true                  |true                            |true                         |
+  +---------------+-------------------------+--------------+---------+-------------------------+----------------------+--------------------------------+-----------------------------+
+  */
+  // scalastyle:on line.size.limit
+  Seq("path", "table").foreach { outputLoc =>
+    Seq("options", "tblproperties", "options_and_tblproperties").foreach { configInput =>
+      Seq("create", "replace", "c_or_r_create", "c_or_r_replace").foreach { sqlOp =>
+        Seq(true, false).foreach { useAsSelectStmt =>
+          val testName = s"SQL - outputLoc=$outputLoc & configInput=$configInput & sqlOp=$sqlOp" +
+            s" & useAsSelectStmt=$useAsSelectStmt"
+
+          test(testName) {
+            withTempDir { dir =>
+              withTable("tbl", "other") {
+                if (sqlOp.contains("replace")) {
+                  var stmt = "CREATE TABLE tbl (ID INT) USING DELTA"
+                  if (outputLoc == "path") {
+                    stmt = stmt + s" LOCATION '${dir.getCanonicalPath}'"
+                  }
+                  sql(stmt)
+                }
+
+                val sqlOpStr = sqlOp match {
+                  case "c_or_r_create" | "c_or_r_replace" => "CREATE OR REPLACE"
+                  case _ => sqlOp.toUpperCase(Locale.ROOT)
+                }
+
+                val schemaStr = if (useAsSelectStmt) "" else "(id INT) "
+                var stmt = sqlOpStr + " TABLE tbl " + schemaStr + "USING DELTA\n"
+
+                if (configInput.contains("options")) {
+                  stmt = stmt + s"OPTIONS('$config_no_prefix'=3,'$config_prefix'=true)\n"
+                }
+                if (outputLoc == "path") {
+                  stmt = stmt + s"LOCATION '${dir.getCanonicalPath}'\n"
+                }
+                if (configInput.contains("tblproperties")) {
+                  stmt = stmt + s"TBLPROPERTIES('$config_no_prefix_2'='interval 60 days'," +
+                    s"'$config_prefix_2'=20)\n"
+                }
+                if (useAsSelectStmt) {
+                  sql("CREATE TABLE other (id INT) USING DELTA")
+                  stmt = stmt + "AS SELECT * FROM other\n"
+                }
+
+                // scalastyle:off println
+                println(stmt)
+                // scalastyle:on println
+                sql(stmt)
+                sql("SHOW TBLPROPERTIES tbl").show(truncate = false)
+
+                val config = sql("SHOW TBLPROPERTIES tbl").collect().map { row =>
+                  val key = row.getString(0)
+                  val value = row.getString(1)
+                  (key, value)
+                }.toMap
```

Comment:
Could you read from `Metadata.configuration` instead? `SHOW TBLPROPERTIES` may not output the same as `Metadata.configuration` IIRC.

Reply:
done